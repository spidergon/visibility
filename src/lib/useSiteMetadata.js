import { useStaticQuery, graphql } from 'gatsby'

const useSiteMetadata = () => {
  const { site } = useStaticQuery(
    graphql`
      query SiteMetaData {
        site {
          siteMetadata {
            title
            description
            author
            keywords
          }
        }
      }
    `
  )
  return site.siteMetadata
}

export default useSiteMetadata
