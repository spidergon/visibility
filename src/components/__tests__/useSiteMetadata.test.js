import { renderHook } from 'react-hooks-testing-library'
import useSiteMetadata from '../../lib/useSiteMetadata'

describe('Hook: useSiteMetadata', () => {
  it('returns the site metadata', () => {
    const { result } = renderHook(() => useSiteMetadata())
    expect(result).toMatchSnapshot()
  })
})
