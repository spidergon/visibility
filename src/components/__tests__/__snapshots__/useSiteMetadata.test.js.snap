// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hook: useSiteMetadata returns the site metadata 1`] = `
Object {
  "current": Object {
    "author": "Christopher Servius",
    "description": "Une description.",
    "keywords": Array [
      "some",
      "keywords",
    ],
    "title": "Site Title",
  },
  "error": undefined,
}
`;
